src/cucumber-json.js

```diff
@@ -1,62 +1,61 @@
-var shell = require('shelljs'),
-config;
+var shell   = require('shelljs'),
+    compose = require('./compose');
 
-function cucumberJSON(d) {
-    return new Promise(function(resolve, reject){
-        var jsonCucumber;
-        var shellOut = shell.exec(
-            command(d),
-            { silent: true, async: false }
-        );
-        if (shellOut.stderr) {
-            reject(shellOut.stderr);
-        }
-        jsonCucumber = JSON.parse(shellOut.stdout);
-        resolve(
-            {
-                [d]: {
-                    testType: config.testType || 'all tests',
-                    count: totalNumberofTests(jsonCucumber),
-                    jsonCucumber: jsonCucumber
-                }
+function fetch(config) {
+    return Promise
+        .all(promises(config))
+        .then(function(arrayOfCucumberJSON) {
+            return arrayOfCucumberJSON;
+        })
+        .catch(function(error) {
+            return error;
+        });
+}
+
+function promises(config) {
+    return config.domains.map(function (domain) {
+        return new Promise(function(resolve, reject) {
+            var process, features;
+
+            process = exec(domain, config);
```

Comment:
@cookiescrumbs I wonder if this promise could be abstracted into an if statement:

```
var result = action();
if (result.stderror) {
    failure(reject, result.stderror);
} else {
    success(resolve, result.stdout);
}
```

But that would mean creating closures for success & failure that are bound to the domain and config variables somewhere else; it would remove the dependencies on domain and config within the promise - but it’s making it more complex with some factory methods for creating the success and failure closures. The failure one would just reject, the success one would resolve with the object.